lib/Dialect/ATen/ATenLayerNamePass.cpp

```diff
@@ -0,0 +1,96 @@
+//===- ATenLayerNamePass.cpp ------------------------------------*- C++ -*-===//
+//
+// This file is licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "npcomp/Dialect/ATen/ATenLayerNamePass.h"
+#include "npcomp/Dialect/ATen/ATenDialect.h"
+
+#include "llvm/Support/Debug.h"
+#include "llvm/Support/ErrorHandling.h"
+#include "llvm/Support/raw_ostream.h"
+
+#include "mlir/Pass/Pass.h"
+
+#include <iostream>
+#include <vector>
+
+#define DEBUG_TYPE "aten-layer-name"
+
+using namespace mlir;
+
+namespace {
+
+struct ATenLayerNamePass
+    : public PassWrapper<ATenLayerNamePass, OperationPass<ModuleOp>> {
+
+private:
+  std::map<Operation *, std::string> opToName;
+
+public:
+  ATenLayerNamePass() {}
+
+  void runOnOperation() override {
+
+    markAllAnalysesPreserved();
+
+    auto module = getOperation();
+
+    // find the function called 'graph'
+    auto graph = module.lookupSymbol<mlir::FuncOp>("graph");
+    if (!graph) {
+      emitError(mlir::UnknownLoc::get(module.getContext()),
+                "OpReportPass failed: can't find a graph function\n");
+      signalPassFailure();
+      return;
+    }
+
+    // Construct a name for each aten operation
+    std::map<std::string, uint64_t> layerIDmap;
+    unsigned currentLayer = 0;
+
+    graph.walk([&](Operation *op) {
+      auto name = op->getName().getStringRef();
+
+      // if it's not an aten operation, continue
+      if (!name.startswith("aten."))
```

Comment:
Maybe a TODO to convert this name munging to an op interface?

Reply:
Moved to issue